policy-controller/k8s/status/src/index.rs

```diff
@@ -21,19 +24,23 @@ pub const STATUS_CONTROLLER_NAME: &str = "status-controller";
 pub type SharedIndex = Arc<RwLock<Index>>;
 
 pub struct Controller {
+    claims: Receiver<Arc<Claim>>,
     client: k8s::Client,
+    name: String,
     updates: UnboundedReceiver<Update>,
+
+    // True if this status controller is the leader — false otherwise.
+    leader: bool,
 }
 
 pub struct Index {
-    // todo: Will be used to compare against the current claim's claimant to
-    // determine if this status controller is the leader
-    _name: String,
-
-    // todo: Will be used in the IndexNamespacedResource trait methods to
-    // check who the current leader is and if updates should be sent to the
-    // controller
-    _claims: Receiver<Arc<Claim>>,
+    // Used to compare against the current claim's claimant to determine if
+    // this status controller is the leader.
+    name: String,
+
+    // Used in the IndexNamespacedResource trait methods to check who the
+    // current leader is and if updates should be sent to the controller.
+    claims: Receiver<Arc<Claim>>,
```

Comment:
Is it worth it to have the Index be claims aware?  As far as I can tell, there should be no harm in allowing the Index to operate normally all of the time, including calling `reconcile` periodically and sending Updates to the controller (which will be dropped if we're not the lease holder).

Reply:
I don't think this is strictly necessary, but it does allow us to skip creating patches and sending them to the `updates` channel when we know we're not the leader. If this is a blocker for you I can think about it more, but I'm inclined to keep it as I don't find it adds that much complexity to the `Index` methods and saves on potentially unnecessary work.

Reply:
It only saves work when the current process is not the leader, so I don't think there's much benefit.  I definitely find it easier to understand how the system works if claims are checked in just one place.

Reply:
So we could remove the `Claim::is_current_for` check that happens in the watched resources traits. We'd always send updates to `Controller`, but we know they'll be dropped anyways.

The other place `claims` is used though is that we call `Index::reconcile` when the claim changes (or 10 seconds elapse), and we do need that to be part of `Index`. When the claim changes, we know `Index` has an up-to-date `http_routes` and `servers` maps, but we want to make sure that the statuses on HTTPRoutes are correct as well. There may have been some missed updates when changing claimants, so `Index::reconcile` needs to send all the updates to `Controller`.

Reply:
Ah yeah, I see what you mean.  Triggering a reconciliation immediately upon acquiring the lease does make a lot of sense.  Thanks for walking me through that.